lib/atmega328/max72xx.cpp

```diff
@@ -30,18 +32,58 @@ static void inline activate_cs(void)
     PORTB &= ~bit(CS_PIN);
 }
 
-void max72xx_write_byte(uint8_t reg, uint8_t data)
+void max72xx_write_byte(uint8_t device, uint8_t reg, uint8_t data)
 {
     activate_cs();
-    SPI_transmit_byte((max72xx_reg_t)reg);
-    SPI_transmit_byte(data);
+    for( uint8_t d = MAX72XX_NUM_DEVICES; d > 0; d--)
+    {
+        if ( d == device)
+        {
+            SPI_transmit_byte(reg);
+            SPI_transmit_byte(data);
+        }
+        else
+        {
+            SPI_transmit_byte(Max72XX_Command_Nop);
+            SPI_transmit_byte(0x00);
+        }
+    }
     deactivate_cs();
 }
 
 void max72xx_clear(void)
 {
-    for (uint8_t i = 0; i < 8; i++)
+    for(uint8_t device = 0; device < MAX72XX_NUM_DEVICES; device++)
+    {
+        for (uint8_t i = 0; i < 8; i++)
+        {
+            max72xx_write_byte(device, Max72XX_Digit0 + i, 0x00);
+        }
+    }
+}
+
+void max72xx_set_intensity(uint8_t intensity)
+{
+    if(intensity > 0x0F) intensity = 0x0F;
+
+    for (uint8_t device = 0; device < MAX72XX_NUM_DEVICES; device++)
+    {
+        max72xx_write_byte(device, Max72XX_Intensity, intensity);
+    }
+}
```

Comment:
There's a more efficient way of writing to all at once:

```suggestion
void max72xx_send_command_to_all(max72xx_cmd_t cmd) {
    max72xx_cmd_t cmds[MAX72XX_NUM_DEVICES];
    for (int i=0; i < MAX72XX_NUM_DEVICES; i++) {
        memcpy(cmds + i, &cmd, sizeof(cmd));
    }

    max72xx_send_commands(cmds, MAX72XX_NUM_DEVICES);
}

void max72xx_set_intensity(uint8_t intensity)
{
    if(intensity > 0x0F) intensity = 0x0F;
    max72xx_send_command_to_all(Max72XX_Intensity, intensity);
}
``

Reply:
I think I managed this.